### PR TITLE
fix CCParticleSystem error for fireball/issues/5155

### PR DESCRIPTION
**Pull request by knoxHuang**
Re: cocos-creator/fireball#5155

Changes proposed in this pull request:
 * 

> Makes sure these boxes are checked before submitting your PR - thank you!
>
- [ ] If your pull request has gone "stale", you should **rebase** your work on top of the latest version of the upstream branch.
- [ ] If your commit history is full of small, unimportant commits (such as "fix pep8" or "update tests"), **squash** your commits down to a few, or one, discreet changesets before submitting a pull request.
- To official teams:
  - [ ] Check that your javascript is following our [style guide](https://github.com/cocos-creator/fireball/blob/dev/.github/CONTRIBUTING.md) and end files with a newline
  - [ ] Document new code with comments in source code based on [API Docs](https://github.com/cocos-creator/fireball#api-docs)
  - [ ] Make sure any log information in `cc.log` , `cc.error`, `cc.warn` or `cc.assert` has been moved into `DebugInfos.js` with an ID

@cocos-creator/engine-admins

### CODE REVIEW

**Comment by jareguo** [comment]
能分别解释下吗？

File: cocos2d/particle/CCParticleSystem.js [hunk 1]
```diff
@@ -912,8 +912,14 @@ var ParticleSystem = cc.Class({
                 sgNode.particleCount = 0;
 
                 var active = sgNode.isActive();
-                sgNode._plistFile = file;
-                sgNode.initWithDictionary(content, '');
+
+                if (CC_EDITOR) {
+                    sgNode._plistFile = file;
+                    sgNode.initWithDictionary(content, '');
```
Comment (pandamicro) [inline cocos2d/particle/CCParticleSystem.js:918]: 同时加载两次相同的粒子系统的情况，Editor 环境下如果用 initWithFile 时，会导致第二次粒子系统 loader 找不到 file（被第一次主动删除），导致第二次粒子初始化报错，所以这里直接使用加载的结果 content。

设置 _plistFile 是为了能够找到索引的独立贴图地址

而在 native 环境下，由于不能够设置 _plistFile 所以用 initWithFile（运行时不会删除缓存，所以同时加载也不会出错）

File: cocos2d/particle/CCSGParticleSystem.js [hunk 2]
```diff
@@ -70,7 +70,7 @@ cc.Particle = function (pos, startPos, color, deltaColor, size, deltaSize, rotat
     this.pos = pos || cc.v2(0, 0);
     this.startPos = startPos || cc.v2(0, 0);
     this.color = color || cc.color(0, 0, 0, 255);
-    this.deltaColor = deltaColor || cc.color(0, 0, 0, 255);
+    this.deltaColor = deltaColor || {r:0, g: 0, b:0, a:255};
```
Comment (pandamicro) [inline cocos2d/particle/CCSGParticleSystem.js:73]: deltaColor 是粒子系统中唯一可能在各个通道有负值的情况，新的 color 类型已经不支持负值了，所以用这种方式取代

**Review by pandamicro — APPROVED** [review]

File: cocos2d/particle/CCSGParticleSystem.js (hunk 2 above)
Comment (jareguo) [reply cocos2d/particle/CCSGParticleSystem.js:73]: 嗯…… 粒子这里之后可以改用 ColorF 类型

File: cocos2d/particle/CCParticleSystem.js (hunk 1 above)
Comment (jareguo) [reply cocos2d/particle/CCParticleSystem.js:918]: 明白了

**Pull request merged**